Sources/NIOSSL/SSLCertificate.swift

```diff
@@ -434,3 +434,68 @@ internal class SubjectAltNameSequence: Sequence, IteratorProtocol {
         CNIOBoringSSL_GENERAL_NAMES_free(self.nameStack)
     }
 }
+
+extension NIOSSLCertificate: CustomStringConvertible {
+    
+    public var description: String {
+        var desc = "<NIOSSLCertificate"
+        if let commonNameBytes = self.commonName() {
+            let commonName = String(decoding: commonNameBytes, as: UTF8.self)
+            desc += ";common_name=" + commonName
+        }
+        if let alternativeName = self.subjectAlternativeNames() {
+            desc += ";alternative_names="
+            for altName in alternativeName {
+                switch altName {
+                case .dnsName(let bytes):
+                    desc += "\(String(decoding: bytes, as: UTF8.self)),"
+                case .ipAddress(let address):
+                    desc += "\(address.description),"
+                }
+            }
+        }
+        return desc + ">"
+    }
+    
+}
+
+extension NIOSSLCertificate.IPAddress: CustomStringConvertible {
+    
+    private static let ipv4AddressLength = 16
+    private static let ipv6AddressLength = 46
+    
+    /// A string representaion of the IP address.
+    /// E.g. IPv4: `192.168.0.1`
+    /// E.g. IPv6: `2001:db8::1`
+    public var description: String {
+        switch self{
+        case .ipv4(let addr):
+            return self.ipv4ToString(addr)
+        case .ipv6(let addr):
+            return self.ipv6ToString(addr)
+        }
+    }
+    
+    private func ipv4ToString(_ address: in_addr) -> String {
+        var address = address
+        var dest: [CChar] = []
+        dest.reserveCapacity(NIOSSLCertificate.IPAddress.ipv4AddressLength)
```

Comment:
This won't work how you want: `withUnsafeMutableBufferPointer` won't include the trailing capacity. We need to zero-fill `dest`: `var dest: [CChar] = Array(repeating: 0, count: ipv4AddressLength)`.